lib/improver/nowcasting/lightning.py

```diff
@@ -0,0 +1,482 @@
+
+            lightning_rate_cube (iris.cube.Cube):
+                Nowcast lightning rate
+                Must have same dimensions as cube
+
+            prob_precip_cube (iris.cube.Cube):
+                Nowcast precipitation probability (threshold > 0.5, 7, 35)
+                Must have same other dimensions as cube
+
+            prob_vii_cube (iris.cube.Cube):
+                Radar-derived vertically integrated ice content (VII)
+                Must have same x and y dimensions as cube
+                Time should be a scalar coordinate
+                Must have a threshold coordinate with points matching
+                self.vii_thresholds
+                Can be <No cube> or None or anything that evaluates to False
+
+        Returns:
+            new_prob_lightning_cube (iris.cube.Cube):
+                Output cube containing Nowcast lightning probability.
+
+        Raises:
+            iris.exceptions.ConstraintMismatchError:
+                If lightning_rate_cube or first_guess_lightning_cube do not
+                contain the expected times.
+        """
+        new_cube_list = iris.cube.CubeList([])
+        # Loop over required forecast validity times
+        for cube_slice in cube.slices_over('time'):
+            this_point = cube_slice.coord('time').points[0]
+            this_time = iris_time_to_datetime(
+                cube_slice.coord('time').copy())[0]
+            prob_lightning_slice = lightning_rate_cube.extract(
+                iris.Constraint(time=this_time))
+            first_guess_time = iris_time_to_datetime(
+                first_guess_lightning_cube.coord('time').copy())[
+                    first_guess_lightning_cube.coord(
+                        'time'
+                        ).nearest_neighbour_index(this_point)]
+            first_guess_slice = first_guess_lightning_cube.extract(
+                iris.Constraint(time=first_guess_time))
+            err_string = "No matching {} cube for {}"
+            if not isinstance(prob_lightning_slice,
+                              iris.cube.Cube):
+                raise ConstraintMismatchError(
+                    err_string.format("lightning", this_time))
+            # Fail if the closest available "first guess" forecast is more
+            # than 2 hours away from the required validity time.
+            if abs((first_guess_time - this_time).total_seconds()) > 7201.:
+                raise ConstraintMismatchError(
+                    err_string.format("first-guess", this_time))
+            cube_slice.data = first_guess_slice.data
+            cube_slice.coord('forecast_period').convert_units('minutes')
+            fcmins = cube_slice.coord('forecast_period').points[0]
+
+            # Increase prob(lightning) to Risk 2 (pl_dict[2]) when within
+            #   lightning halo (lrt_lev2; 50km of an observed ATDNet strike):
+            cube_slice.data = np.where(
+                (prob_lightning_slice.data >= self.lrt_lev2) &
```

Comment:
I'm not sure if you've tested the case where the lightning rate is high but the first guess is also high, so this doesn't change the result

Reply:
Added.